### PR TITLE
Created User registraion, userlist and profile photo upload.

### PR DESCRIPTION
**Pull request by Joyel42**

### CODE REVIEW

File: chat-app-express/src/model/userModel.js [hunk 1]
```diff
@@ -19,7 +16,7 @@ const userSchema = new Schema({
         minLength: [3, 'Name should be at least 3 charcheters']
     },
     phone: {
-        type: Number,
+        type: String,
```
Comment (MONSTER-HARSH) [inline chat-app-express/src/model/userModel.js:19]: why phone number is string?

File: chat-app-express/src/public/images/users/user-018f9450-f79f-703d-a576-db81fe09bf5e-1716186297590.jpeg
Comment (MONSTER-HARSH) [inline chat-app-express/src/public/images/users/user-018f9450-f79f-703d-a576-db81fe09bf5e-1716186297590.jpeg:1]: why is this image?

**Review by MONSTER-HARSH — CHANGES_REQUESTED** [review]

File: chat-app-express/src/model/userModel.js (hunk 1 above)
Comment (Joyel42) [reply chat-app-express/src/model/userModel.js:19]: Some node issues Number can't be parsed into DB

File: chat-app-express/src/public/images/users/user-018f9450-f79f-703d-a576-db81fe09bf5e-1716186297590.jpeg
Comment (Joyel42) [reply chat-app-express/src/public/images/users/user-018f9450-f79f-703d-a576-db81fe09bf5e-1716186297590.jpeg:1]: I kept it for sample testing profile

Comment (Joyel42) [reply chat-app-express/src/public/images/users/user-018f9450-f79f-703d-a576-db81fe09bf5e-1716186297590.jpeg:1]: Wheather in FE images are available. We will delete it later.

File: chat-app-express/src/model/userModel.js (hunk 1 above)
Comment (Joyel42) [reply chat-app-express/src/model/userModel.js:19]: Since we keep validaor its fine

File: chat-app-express/app.js [hunk 2]
```diff
@@ -4,31 +4,33 @@ var path = require('path');
...
-app.use(express.urlencoded({ extended: false }));
 app.use(cookieParser());
-app.use(express.static(path.join(__dirname, 'public')));
+app.use(express.static('src/public'));
 
 app.use('/api/', indexRouter);
 app.use('/api/users', usersRouter);
 
+// DB connection
+mongoose.connect(process.env.DBLINK)
```
Comment (95RahulKumar) [inline chat-app-express/app.js]: make a separate file for mongodb connect (connect.js) export connection function and use it here

File: chat-app-express/app.js [hunk 3]
```diff
@@ -4,31 +4,33 @@ var path = require('path');
...
+  .then(() => console.log('MongoDB connected'))
+  .catch(err => console.error(err));
+
+
 // Allowing all orgins
```
Comment (95RahulKumar) [inline chat-app-express/app.js]: use npm cors  package for allow all origin or specific ip

File: chat-app-express/app.js [hunk 4]
```diff
@@ -38,32 +40,29 @@ app.use(function(req, res, next) {
   next();
 });
 
-// DB connetion
-mongoose.connect(process.env.DBLINK);
-
 io.on('connection', (socket) => {
```
Comment (95RahulKumar) [inline chat-app-express/app.js]: create a separate file for socket connection

**Review by 95RahulKumar — CHANGES_REQUESTED** [review]

File: chat-app-express/src/services/userService.js [hunk 5]
```diff
@@ -1,7 +1,88 @@
...
 
-module.exports = decodeToken
\ No newline at end of file
+const userRegister = async (userDetails,res)=>{
+   try {
+      const userID = generateUserId();
+      const name = userDetails.name;
+      const phone = parseInt(userDetails.phone);
+      const createdAt = new Date();
+      const avatar = "src/public/images/users/default.jpeg";
```
Comment (95RahulKumar) [inline chat-app-express/src/services/userService.js]: if new user then how u are sending image url to FE

Comment (95RahulKumar) [reply chat-app-express/src/services/userService.js]: ok below i seen u are updating the avatar using $set query

File: chat-app-express/app.js (hunk 4 above)
Comment (Joyel42) [reply chat-app-express/app.js]: Done

File: chat-app-express/app.js (hunk 3 above)
Comment (Joyel42) [reply chat-app-express/app.js]: Done

File: chat-app-express/app.js (hunk 2 above)
Comment (Joyel42) [reply chat-app-express/app.js]: Done

File: chat-app-express/app.js [hunk 6]
```diff
@@ -17,36 +16,20 @@ const socket = require("socket.io");
 const server = http.createServer(app);
 const io = socket(server);
 
+var cors = require('cors')
+var dbConnect = require('./src/connect');
+var socketConnect = require('./src/socket')
+
+dbConnect.initiateDBConnection();
```
Comment (95RahulKumar) [inline chat-app-express/app.js:23]: good

File: chat-app-express/app.js [hunk 7]
```diff
@@ -17,36 +16,20 @@ const socket = require("socket.io");
...
+socketConnect.socketConnection(io);
```
Comment (95RahulKumar) [inline chat-app-express/app.js:24]: good

File: chat-app-express/src/controllers/userController.js [hunk 8]
```diff
@@ -1,15 +1,93 @@
...
+    const ext = file.mimetype.split('/')[1];
+    const userId = req.body.userID;
+    const filename = userId ? `user-${userId}-${Date.now()}.${ext}` : `${Date.now()}.${ext}`;
+    let avatar = "src/public/images/users/" + filename
+    const result = await User.updateOne({ userID: userId }, { $set: { avatar: avatar } });
+    cb(null, filename);
+  }
+});
+
+const multerFilter = (req, file, cb) => {
```
Comment (95RahulKumar) [inline chat-app-express/src/controllers/userController.js]: make it middleware

**Review by 95RahulKumar — CHANGES_REQUESTED** [review]

File: chat-app-express/src/controllers/userController.js (hunk 8 above)
Comment (Joyel42) [reply chat-app-express/src/controllers/userController.js]: Done

File: chat-app-express/src/services/userService.js (hunk 5 above)
Comment (Joyel42) [reply chat-app-express/src/services/userService.js]: Ys and it will be in http://172.16.120.37:3000/images/users/example.jpeg

**Review by 95RahulKumar — APPROVED** [review]

**Comment by MONSTER-HARSH** [comment]
done

**Review by MONSTER-HARSH — APPROVED** [review]

**Pull request merged**